modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/cell_renderers/ActionLinkRenderer.js

```diff
@@ -96,9 +96,14 @@ function ActionLinkRenderer({actions, itemData, itemId, options, value}) {
 			else if (currentAction.onClick) {
 				event.preventDefault();
 
-				event.target.setAttribute('onClick', currentAction.onClick);
-				event.target.onclick();
-				event.target.removeAttribute('onClick');
+				if (typeof currentAction.onClick === 'function') {
+					currentAction.onClick({itemData});
+				}
+				else {
+					event.target.setAttribute('onClick', currentAction.onClick);
+					event.target.onclick();
+					event.target.removeAttribute('onClick');
+				}
```

Comment:
Nice and safe! Maybe even a tad too safe for my liking. I would just change this, to drop support for stringified handler. But it's more than fine like this :+1: